src/gtest/test_checktransaction.cpp

```diff
@@ -0,0 +1,26 @@
+#include <gtest/gtest.h>
+
+#include "main.h"
+#include "primitives/transaction.h"
+#include "consensus/validation.h"
+
+TEST(checktransaction_tests, check_vpub_not_both_nonzero) {
+    CMutableTransaction tx;
+    tx.nVersion = 2;
+
+    {
+        // Ensure that values within the pour are well-formed.
+        CMutableTransaction newTx(tx);
+        CValidationState state;
+        state.SetPerformPourVerification(false); // don't verify the snark
```

Comment:
See #862 for the ticket turning this into a mock-based test and removing the `SetPerformPourVerification` API in the future. (*Way* out of scope for this PR.)